packages/open-api-gateway/src/project/smithy/components/sample-executable.ts

```diff
@@ -0,0 +1,38 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+import * as fs from "fs";
+import * as path from "path";
+import { Project, SampleFile, SampleFileOptions } from "projen";
+import { getFilePermissions } from "projen/lib/util";
+
+/**
+ * A sample file that is executable
+ */
+export class SampleExecutable extends SampleFile {
+  private readonly file: string;
+
+  constructor(project: Project, filePath: string, options: SampleFileOptions) {
+    super(project, filePath, options);
+    this.file = filePath;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public synthesize() {
+    const fullFilePath = path.join(this.project.outdir, this.file);
```

Comment:
Should we just store this in the private filePath variable as part of the constructor?

Reply:
Can do! :)

Reply:
Will name it `fullFilePath` and kill 2 birds with one stone :)